SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UnresolvedReferenceGymTest/UnresolvedReferenceGymTest.cpp

```diff
@@ -0,0 +1,57 @@
+
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "UnresolvedReferenceGymTest.h"
+
+AUnresolvedReferenceGymTest::AUnresolvedReferenceGymTest()
+	: Super()
+{
+	Author = "Ajanth";
+	Description = TEXT("Tests what happens when structs with references to actors whose entity have not been created yet are replicated.");
+}
+
+void AUnresolvedReferenceGymTest::PrepareTest()
+{
+	Super::PrepareTest();
+
+	AddStep(TEXT("MakeActorReferences"), FWorkerDefinition::Server(1), nullptr, [this]() {
+		AUnresolvedReferenceGymTestActor* ActorRefsActor =
+			GetWorld()->SpawnActor<AUnresolvedReferenceGymTestActor>({ 0.0f, 0.0f, 0.0f }, FRotator::ZeroRotator);
+		RegisterAutoDestroyActor(ActorRefsActor);
+
+		for (int i = 0; i < 3; i++)
+		{
+			AReplicatedTestActorBase* Actor =
+				GetWorld()->SpawnActor<AReplicatedTestActorBase>(FVector(0, 0, 0), FRotator::ZeroRotator, FActorSpawnParameters());
+			(ActorRefsActor->ActorRefs).Add(Actor);
+			RegisterAutoDestroyActor(Actor);
+		}
+
+		FinishStep();
+	});
+
+	AddStep(TEXT("CheckNoNullReferences"), FWorkerDefinition::AllClients, nullptr, nullptr, [this](float deltaTime) {
+		for (TActorIterator<AUnresolvedReferenceGymTestActor> Iter(GetWorld()); Iter; ++Iter)
+		{
+			int maxAllowedRepNotifies = 15;
```

Comment:
Should probably be defined outside the loop somewhere either as a static on the test class or in an anonymous namespace at the top of the cpp.

Reply:
I don't see why this can't be integer literal at the site of the comparison.
E.g.
```
RequireCompare_Int(Iter->numRepNotify, EComparisonMethod::Less_Than_Or_Equal_To, 15, TEXT("Should have gotten no more than 15 RepNotifies."));
```

Reply:
I also don't understand why we expect this number to vary? Is the amount of RepNotifies non deterministic?

Reply:
Yeah that sounds nicer.

Reply:
+1 for a timer, rep notifies would likely be irrelevant.